**Proyecto Waze: análisis de hipótesis sobre el tipo de dispositivo y la cantidad de viajes**

Metodología PACE

Este análisis sigue el marco PACE (Plan, Analyze, Construct, Execute) para un enfoque estructurado y orientado a resultados.

1. PACE: Planificar

Pregunta de investigación: ¿Existe una diferencia en el número promedio de viajes entre los usuarios de iPhone y los usuarios de Android en la aplicación Waze?

Hipótesis nula (H0): No hay diferencia en el número promedio de viajes entre los dos grupos.

Hipótesis alternativa (H1): Existe una diferencia en el número promedio de viajes entre los dos grupos.

2. PACE: Analizar y Construir

Carga de datos e importación de librerías:

In [2]:
import pandas as pd
from scipy import stats

df = pd.read_csv('waze_dataset.csv')

Exploración de datos y estadísticas descriptivas:

In [3]:
# Convertir la variable 'device' a numérica
map_dictionary = {'Android': 2, 'iPhone': 1}
df['device_type'] = df['device'].map(map_dictionary)

# Calcular el promedio de viajes por tipo de dispositivo
average_drives_by_device = df.groupby('device_type')['drives'].mean()
print(average_drives_by_device)

device_type
1    67.859078
2    66.231838
Name: drives, dtype: float64


Prueba de hipótesis (t-test de dos muestras independientes):

In [4]:
iphone_drives = df[df['device_type'] == 1]['drives']
android_drives = df[df['device_type'] == 2]['drives']

t_statistic, p_value = stats.ttest_ind(a=iphone_drives, b=android_drives, equal_var=False)
print(f'Estadístico t: {t_statistic}')
print(f'Valor p: {p_value}')

Estadístico t: 1.463523206885235
Valor p: 0.143351972680206


Interpretación del valor p: dado que el valor p (0.143) es mayor que el nivel de significancia del 5% (0.05), no se rechaza la hipótesis nula.

3. PACE: Ejecutar

Resultados: no se encontró una diferencia estadísticamente significativa en el número promedio de viajes entre los usuarios de iPhone y los usuarios de Android.

Recomendaciones:
explorar otros factores que puedan influir en la variación del número de viajes, como la antigüedad del usuario, la frecuencia de uso de la aplicación y la demografía del usuario.

Realizar pruebas de hipótesis adicionales para investigar otras posibles diferencias en el comportamiento de los usuarios.

Implementar cambios temporales en el marketing o la interfaz de usuario de la aplicación Waze para recopilar más datos y comprender mejor el comportamiento de los usuarios.

Conclusión

El análisis estadístico realizado no encontró evidencia suficiente para afirmar que existe una diferencia significativa en el número promedio de viajes entre los usuarios de iPhone y los usuarios de Android en la aplicación Waze. Se recomienda realizar análisis adicionales para explorar otros factores que puedan influir en el comportamiento de los usuarios y mejorar las estrategias de retención.